sentry/src/routes/channel.rs

```
@@ -342,6 +336,74 @@ pub async fn get_spender_limits<A: Adapter + 'static>(
+        .extensions()
+        .get::<ChannelV5>()
+        .expect("Request should have Channel")
+        .to_owned();
+
+    let new_state = match get_corresponding_new_state(&app.pool, &channel).await? {
+        Some(new_state) => new_state,
+        None => return Err(ResponseError::NotFound),
+    };
+
+    let mut all_spender_limits: HashMap<Address, Spender> = HashMap::new();
+
+    let all_spendables = get_all_spendables_for_channel(app.pool.clone(), &channel.id()).await?;
+
+    // Using for loop to avoid async closures
+    for spendable in all_spendables {
+        let spender = spendable.spender;
+        let spender_leaf = if new_state.msg.balances.spenders.contains_key(&spender) {
+            let balance = new_state.msg.balances.spenders.get(&spender).unwrap();
+
+            Some(SpenderLeaf {
+                total_spent: spendable
+                    .deposit
+                    .total
+                    .checked_sub(balance)
+                    .unwrap_or_default(),
+                // merkle_proof: [u8; 32], // TODO
+            })
+        } else {
+            None
+        };
+
+        let spender_info = Spender {
+            total_deposited: spendable.deposit.total,
+            spender_leaf,
+        };
+
+        all_spender_limits.insert(spender, spender_info);
+    }
+
+    let res = AllSpendersResponse {
+        spenders: all_spender_limits,
+    };
+
+    Ok(success_response(serde_json::to_string(&res)?))
+}
+
+async fn get_corresponding_new_state(
+    pool: &DbPool,
+    channel: &ChannelV5,
+) -> Result<Option<MessageResponse<NewState<UncheckedState>>>, ResponseError> {
+    let approve_state = match latest_approve_state_v5(pool, channel).await? {
+        Some(approve_state) => approve_state,
+        None => return Err(ResponseError::NotFound),
+    };
+
+    let state_root = approve_state.msg.state_root.clone();
+
+    let new_state = latest_new_state_v5(pool, channel, &state_root).await?;
+
+    Ok(new_state)
+}
```

we should change this function to align with the Channel cases:

If there is no `ApproveState` - then we should return `None`. Since the channel might be new, there might not be an `ApproveState` yet.

If there is an `ApproveState` then we must fetch the `NewState`, however, since the `ApproveState` approves a given `NewState`, if `NewState` is not found we should return an error. It should not be a `NotFound` but a more critical one like `BadRequest`.